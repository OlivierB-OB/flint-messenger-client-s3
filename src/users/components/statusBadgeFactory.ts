import Badge from '@material-ui/core/Badge';
import withStyles from '@material-ui/core/styles/withStyles';
import { red, grey, green } from '@material-ui/core/colors';
import { IUserStatus } from '../../identity/types';

const cache = new Map();

export default function statusBadgeFactory(status: IUserStatus) {
  if (cache.has(status)) return cache.get(status);
  const color = status === 'available' ? green[500] : status === 'incall' ? red[500] : grey[500];
  const StatusBadge = withStyles(() => ({
    badge: {
      backgroundColor: color,
      color: color,
    },
  }))(Badge);
  cache.set(status, StatusBadge);
  return StatusBadge;
}
