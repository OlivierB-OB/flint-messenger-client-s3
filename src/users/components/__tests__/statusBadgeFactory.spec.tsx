import React from 'react';
import { shallow } from 'enzyme';
import statusBadgeFactory from '../statusBadgeFactory';

describe('statusBadgeFactory', () => {
  it('available', async () => {
    const Badge = statusBadgeFactory('available');
    const component = shallow(<Badge />);
    expect(component).toMatchSnapshot();
  });

  it('incall', async () => {
    const Badge = statusBadgeFactory('incall');
    const component = shallow(<Badge />);
    expect(component).toMatchSnapshot();
  });

  it('offline', async () => {
    const Badge = statusBadgeFactory('offline');
    const component = shallow(<Badge />);
    expect(component).toMatchSnapshot();
  });
});
