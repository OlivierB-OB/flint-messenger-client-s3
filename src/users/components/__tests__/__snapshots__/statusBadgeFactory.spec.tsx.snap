// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statusBadgeFactory available 1`] = `
<WithStyles(ForwardRef(Badge))
  classes={
    Object {
      "badge": "WithStyles(ForwardRef(Badge))-badge-1",
    }
  }
/>
`;

exports[`statusBadgeFactory incall 1`] = `
<WithStyles(ForwardRef(Badge))
  classes={
    Object {
      "badge": "WithStyles(ForwardRef(Badge))-badge-2",
    }
  }
/>
`;

exports[`statusBadgeFactory offline 1`] = `
<WithStyles(ForwardRef(Badge))
  classes={
    Object {
      "badge": "WithStyles(ForwardRef(Badge))-badge-3",
    }
  }
/>
`;
